internal/server/helper.go

```diff
@@ -1,33 +1,69 @@
 package server
 
 import (
-	"log"
-
 	"github.com/SOMAS2020/SOMAS2020/internal/common"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/config"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 )
 
-func getClientInfoFromRegisteredClients(registeredClients map[shared.ClientID]common.Client) map[shared.ClientID]common.ClientInfo {
+func getClientInfosAndMapFromRegisteredClients(
+	registeredClients map[shared.ClientID]common.Client,
+) (map[shared.ClientID]common.ClientInfo, map[shared.ClientID]common.Client) {
 	clientInfos := map[shared.ClientID]common.ClientInfo{}
+	clientMap := map[shared.ClientID]common.Client{}
 
 	for id, c := range registeredClients {
 		clientInfos[id] = common.ClientInfo{
-			Client:    c,
-			Resources: common.DefaultResources,
-			Alive:     true,
+			Resources:                    config.InitialResources,
+			Alive:                        true,
+			Critical:                     false,
+			CriticalConsecutiveTurnsLeft: config.MaxCriticalConsecutiveTurns,
 		}
+		clientMap[id] = c
 	}
 
-	return clientInfos
+	return clientInfos, clientMap
 }
 
-// anyClientsAlive returns true if any client is alive in the provided ClientInfos map.
+// anyClientsAlive returns true if any one client is Alive (including critical).
 func anyClientsAlive(clientInfos map[shared.ClientID]common.ClientInfo) bool {
-	for id, ci := range clientInfos {
+	for _, ci := range clientInfos {
 		if ci.Alive {
-			log.Printf("%v", id)
 			return true
 		}
 	}
 	return false
 }
+
+// updateIslandLivingStatusForClient returns an updated copy of the clientInfo after updating
+// the Alive, Critical, and CriticalConsecutiveTurnsLeft attribs according to the resource levels and
+// the game's configuration.
+func updateIslandLivingStatusForClient(ci common.ClientInfo, minimumResourceThreshold int, MaxCriticalConsecutiveTurns uint) common.ClientInfo {
+	// we don't resurrect dead clients!
+	if !ci.Alive {
+		return ci
+	}
+

```

Comment:
Do we want events for things like entering/leaving critical status that might be of use to the client?

Reply:
The client should track that itself